src/Adapter/Module/CommandHandler/ResetModuleHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Module\CommandHandler;
+
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Module\Command\ResetModuleCommand;
+use PrestaShop\PrestaShop\Core\Domain\Module\CommandHandler\ResetModuleHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Module\Exception\CannotResetModuleException;
+use PrestaShop\PrestaShop\Core\Module\ModuleManager;
+use PrestaShop\PrestaShop\Core\Module\ModuleRepository;
+
+#[AsCommandHandler]
+class ResetModuleStatusHandler implements ResetModuleHandlerInterface
+{
+    public function __construct(
+        protected ModuleManager $moduleManager,
+        protected ModuleRepository $moduleRepository,
+    ) {
+    }
+
+    public function handle(ResetModuleCommand $command): void
+    {
+        $this->moduleRepository->getPresentModule($command->getTechnicalName()->getValue());
+
+        if (!$this->moduleManager->isInstalledAndActive($command->getTechnicalName()->getValue())) {
+            throw new CannotResetModuleException('Module not installed or not enabled', CannotResetModuleException::NOT_ACTIVE);
+        }
```

Comment:
The previous test was more accurate it tested both uninstalled and disabled states instead of both at the same time, we can also use the returned object from the method as the reference for this kind of checks instead of the module manager that hides the logic

The `Module` object already contains all the info we need so I think it should be our reference in all the CQRS handlers

Reply:
```suggestion
        $module = $this->moduleRepository->getPresentModule($command->getTechnicalName()->getValue());

        if (!$module->isInstalled()) {
            throw new CannotResetModuleException('Cannot reset a module not installed', CannotResetModuleException::NOT_INSTALLED);
        }
        
        if (!$module->isActive()) {
            throw new CannotResetModuleException('Cannot reset a disabled module', CannotResetModuleException::NOT_ACTIVE);
        }
```